internal/ingest/linksystem.go

```diff
@@ -172,101 +146,24 @@ func providerFromAd(ad schema.Advertisement) (peer.ID, error) {
 // hook is being called for each entry in an advertisement's chain of entries.
 // Process the entry and save all the multihashes in it as indexes in the
 // indexer-core.
-func (ing *Ingester) storageHook(pubID peer.ID, c cid.Cid) {
-	log := log.With("publisher", pubID, "cid", c)
+func (ing *Ingester) ingestEntryChunk(publisher peer.ID, adCid cid.Cid, entryChunkCid cid.Cid) error {
+	log := log.With("publisher", publisher, "adCid", adCid, "cid", entryChunkCid)
 
 	// Get data corresponding to the block.
-	val, err := ing.ds.Get(context.Background(), dsKey(c.String()))
+	val, err := ing.ds.Get(context.Background(), dsKey(entryChunkCid.String()))
 	if err != nil {
 		log.Errorw("Error while fetching the node from datastore", "err", err)
-		return
+		return err
```

Comment:
Please do not log the error, and then also return it (presumably to be logged again).  Prefer to log at the highest level so that the message can contain as much info as possible.
```suggestion
		return fmt.Errorf("cannot fetch node from datastore: %w", err)
```